aw_core/timeperiod.py

```diff
@@ -46,3 +67,21 @@ def intersection(self, other: "TimePeriod") -> Optional["TimePeriod"]:
             # Entirety of self is within other
             return self
         return None
+
+    def adjacent(self, other: "TimePeriod") -> bool:
+        """Iff timeperiods are exactly next to each other, return True."""
+        return self.start == other.end or self.end == other.start
+
+    def gap(self, other: "TimePeriod") -> Optional["TimePeriod"]:
+        """If periods are separated by a non-zero gap, return the gap as a new timeperiod, else None"""
+        if not (self.overlaps(other) or self.adjacent(other)):
```

Comment:
Won't a simple if statement be enough here?
```python
if self.end < other.start:
    return TimePeriod(self.end, other.start)
elif other.end < self.start:
    return TimePeriod(other.end, self.start)
else: 
    return None
```

Reply:
Yeah that would be cleaner! Thanks, fixed!